saleor/warehouse/management.py

```diff
@@ -36,11 +39,15 @@ def allocate_stocks(
         return
 
     variants = [line_info.variant for line_info in order_lines_info]
+    filter_lookup = {"product_variant__in": variants}
+
+    if additional_filter_lookup is not None:
+        filter_lookup.update(additional_filter_lookup)
 
     stocks = list(
         Stock.objects.select_for_update(of=("self",))
         .for_country_and_channel(country_code, channel_slug)
```

Comment:
This requirement is too strict for current click and collect implementation.
We do not take into the account shipping address, thus, we cannot (or do not want)
To calculate warehouses/stocks for given country. It is possible to extend
 available collection points by country/channel filter, but we need to know shipping
(or user) address